nucypher/utilities/porter/control/specifications/porter_schema.py

```diff
@@ -159,5 +159,18 @@ class BobRetrieveCFrags(BaseSchema):
             type=click.STRING,
             required=True))
 
+    # optional
+    context = base_fields.Base64JSON(
```

Comment:
Just what I needed :ok_hand:

Reply:
@piotr-roslaniec: Question for you (need your web expertise).

One of the reasons I created Base64JSON was to allow for URL request parameters. Does this make sense to do? We can't pass nested dicts as URL parameters so if the context JSON had nested dicts in it, that would be problematic. The Base64 encoding takes care of handling this.

Interested to get your thoughts. Should we support URL parameters? Porter did before.

Reply:
@derekpierre 
IIRC the original reason for using URL-encoded parameters was that the browsers don't like `GET` requests with JSON body. Base64JSON makes sense because it solves the issue of sending complex types (dictionaries) as parameters in URLs. And since context etc. is never going to be used on its own etc., the fact that the context type and structure are obfuscated (base64-encoded) will not cause any issues.

Reply:
> IIRC the original reason for using URL-encoded parameters was that the browsers don't like GET requests with JSON body

Does it change your mind at all given that the retrieve endpoint is a `POST` and not a `GET`?

Reply:
Well, actually, yes. In that case, `Base64JSON` is not _necessary_. 

So I think I should ask, is there any reason why we shouldn't send plain JSON then? Maybe we'd like to obfuscate the condition context because the structure/API will be unstable, so it's better to present it as Base64JSON? 

I'm just throwing ideas, but if there are no objections then I think plain JSON is fine. What do you think @derekpierre ?

Reply:
> Maybe we'd like to obfuscate the condition context because the structure/API will be unstable
IMHO JSON is robust enough to handle instability in the API. I guess any obfuscation of condition context would probably take the form of a proof in the context...maybe (?)

On the simplicity/readability side, plain JSON is good. If there are no need for URL parameters for a POST endpoint, then I'm pro just plain JSON.

cc @KPrasch for his thoughts.

Reply:
FYI - what the code would look like if we switched to plain JSON - https://github.com/nucypher/nucypher/pull/2962/commits/27330b4d5f2e3b59fa61327d46d39551dca1b5e8

Reply:
Opted for plain JSON.